prime-router/src/main/kotlin/tokens/Server2ServerAuthentication.kt

```diff
@@ -189,20 +183,35 @@ class Server2ServerAuthentication(val workflowEngine: WorkflowEngine) : Logging
                         " Unable to find auth key for ${parsedJwt.organization.name} with" +
                         " scope=$scope, kid=${parsedJwt.kid}, and alg=${parsedJwt.kty}"
                 )
+                throw Server2ServerAuthenticationException(Server2ServerError.NO_VALID_KEYS, scope, parsedJwt.issuer)
+            }
+            if (!possibleKeys.any { key -> verifyJwtWithKey(jwsString, key, jtiCache, actionHistory) }) {
+                throw Server2ServerAuthenticationException(Server2ServerError.NO_VALID_KEYS, scope, parsedJwt.issuer)
             }
-            possibleKeys.any { key -> verifyJwtWithKey(jwsString, key, jtiCache, actionHistory) }
         } catch (ex: Server2ServerAuthenticationException) {
             logErr(actionHistory, ex.localizedMessage)
-            false
-        } catch (ex: JwtException) {
+            throw ex
+        } catch (ex: ExpiredJwtException) {
+            logErr(actionHistory, "AccessToken Request Denied: $ex")
+            throw Server2ServerAuthenticationException(Server2ServerError.EXPIRED_TOKEN, scope)
+        } catch (ex: UnsupportedJwtException) {
+            logErr(actionHistory, "AccessToken Request Denied: $ex")
+            throw Server2ServerAuthenticationException(Server2ServerError.UNSIGNED_JWT, scope)
+        } catch (ex: MalformedJwtException) {
+            logErr(actionHistory, "AccessToken Request Denied: $ex")
+            throw Server2ServerAuthenticationException(Server2ServerError.MALFORMED_JWT, scope)
+        } catch (ex: SignatureException) {
             logErr(actionHistory, "AccessToken Request Denied: $ex")
-            false
+            throw Server2ServerAuthenticationException(Server2ServerError.NO_VALID_KEYS, scope)
         } catch (ex: IllegalArgumentException) {
             logErr(actionHistory, "AccessToken Request Denied: $ex")
-            false
+            throw Server2ServerAuthenticationException(Server2ServerError.MALFORMED_JWT, scope)
         } catch (ex: NullPointerException) {
             logErr(actionHistory, "AccessToken Request Denied: $ex")
-            false
+            throw Server2ServerAuthenticationException(
+                Server2ServerError.MALFORMED_JWT,
+                scope
+            )
```

Comment:
I tried looking how to do a multi catch on Kotlin and the only way I found is by using a `when`. Not sure if that will make it any better,  but it avoids having the logErr on every catch. Here's how you could do it: 
catch (ex: Exception) {
    logErr(actionHistory, "AccessToken Request Denied: $ex")
    when(ex) {
        is ExpiredJwtException -> throw Server2ServerAuthenticationException(Server2ServerError.EXPIRED_TOKEN, scope)
        is UnsupportedJwtException ->  throw Server2ServerAuthenticationException(Server2ServerError.UNSIGNED_JWT, scope)
        ....
        else -> throw ex
    }
}